src/Migrator/MigrationWorkflow.cs

```diff
@@ -0,0 +1,157 @@
+                throw new ArgumentOutOfRangeException(nameof(runType), runType, "Invalid MigrationRunner option");
+        }
+    }
+
+    private async Task RunEntryAsync(string[] assemblies, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        var workflowId = Workflow.Info.WorkflowId;
+        var logger = Workflow.Logger;
+
+        if (assemblies.Length == 0)
+        {
+            // scan all domain
+            _loadedTypes.AddRange(AppDomain.CurrentDomain.GetAssemblies().SelectMany(assembly => assembly.GetTypes()));
+        }
+        else
+        {
+            // only load what is needed by configuration.
+            var loadedAssemblies = new List<Assembly>();
+            foreach (var assembly in assemblies)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    var loadedAssembly = Assembly.Load(assembly);
+                    loadedAssemblies.Add(loadedAssembly);
+                }
+                catch (ArgumentNullException)
+                {
+                    logger.LogDebug("{Assembly} string is null", assembly);
+                }
+                catch (ArgumentException)
+                {
+                    logger.LogDebug("{Assembly} string is empty", assembly);
+                }
+                catch (FileNotFoundException)
+                {
+                    logger.LogDebug("{Assembly} could not be found", assembly);
+                }
+                catch (FileLoadException)
+                {
+                    logger.LogDebug("{Assembly} could not be loaded", assembly);
+                }
+                catch (BadImageFormatException)
+                {
+                    logger.LogDebug("{Assembly} corrupted", assembly);
+                }
+            }
+
+            _loadedTypes.AddRange(loadedAssemblies.SelectMany(assembly => assembly.GetTypes()));
+        }
+
+        var migrationTypes = _loadedTypes.Where(t => _migrationType.IsAssignableFrom(t) && t is { IsInterface: false, IsAbstract: false });
+        var richTypes = migrationTypes.Select(t => new { Type = t, Version = t.GetCustomAttribute<MigrationAttribute>()!.Version, });
+        var sortedTypes = richTypes.OrderBy(t => t.Version);
+
+        foreach (var migrationType in sortedTypes.Select(s => s.Type))
+        {
+            var nextType = migrationType.ToString();
+
+            await Workflow.ExecuteChildWorkflowAsync<MigrationWorkflow>(
+                s => s.RunAsync(
+                    RunType.Migration,
+                    new[] { nextType }),
+                new ChildWorkflowOptions
+                {
+                    Id = $"{workflowId}_{migrationType.Name}",
+                    CancellationToken = cancellationToken,
+                    Memo = new Dictionary<string, object>
+                    {
+                        { "Parent", workflowId },
```

Comment:
This should be available already as the Temporal UI keeps track of it.

I don't see a need for it in the memo